exercises/saddle-points/tests/saddle-points.rs

```diff
@@ -0,0 +1,103 @@
+extern crate saddle_points;
+
+use saddle_points::*;
+
+#[test]
+fn test_identify_single_saddle_point() {
+    let vector: Vec<Vec<u64>> = vec![vec![9, 8, 7],
+                                     vec![5, 3, 2],
+                                     vec![6, 6, 7]];
+    let input: &[&[u64]] = &[&vector[0][..],
```

Comment:
This way of forming the input slice of slices is ugly, and is my fault. I inadvertently gave you bad advice. The `find_saddle_points` signature should actually be `pub fn find_saddle_points(_: &[Vec<u64>]) -> Vec<(u64, u64)>`.

I was wrong to tell you that it should accept an `&[&[u64]]`, because the compiler doesn't automatically dereference the inner elements into a slice; the inner elements need to remain `Vec<u64>`. At the same time, the outer layer should be `&[_]` because it is idiomatic to pass a slice reference, not to transfer ownership of a `Vec`. 

In other words, the function signature and tests should work [as in this playground example](https://play.rust-lang.org/?gist=dcc7e8c78a38e261baa4ed8a921e074f&version=stable). 

I apologize for the bad advice. Now please change the signature one more time to hopefully clean this up once and for all.

Reply:
Probably the best function signature for real projects would look like [this](https://play.rust-lang.org/?gist=90f6eb37932f7d5d01afb9515d94eb5b&version=stable):

```rust
pub fn find_saddle_points<X, Y, T>(_: &Y) -> Vec<(T, T)>
where
    Y: AsRef<[X]>,
    X: AsRef<[T]>,
{
    unimplemented!()
}
```

However, that would be an intimidating signature to require of new students. I still prefer `&[Vec<u64>]`.